probe-rs/src/architecture/arm/sequences.rs

```diff
@@ -603,23 +603,35 @@ pub trait ArmDebugSequence: Send + Sync + Debug {
         interface: &mut dyn DapAccess,
         dp: DpAddress,
     ) -> Result<(), ArmError> {
-        // Clear all errors.
-        // CMSIS says this is only necessary to do inside the `if powered_down`, but
-        // without it here, nRF52840 faults in the next access.
-        let mut abort = Abort(0);
-        abort.set_dapabort(true);
-        abort.set_orunerrclr(true);
-        abort.set_wderrclr(true);
-        abort.set_stkerrclr(true);
-        abort.set_stkcmpclr(true);
-        interface.write_dp_register(dp, abort)?;
-
         interface.write_dp_register(dp, SelectV1(0))?;
         let dpidr: DPIDR = interface.read_dp_register(dp)?;
 
+        // Clear all errors and see if we need to power up the target
         let ctrl = interface.read_dp_register::<Ctrl>(dp)?;
+        let powered_down = if dpidr.version() == 0 || dpidr.designer() == 0 {
+            // Abort the current transaction. This is the only bit present in `Abort` in `DPv1`.
```

Comment:
I feel like there is a mix of DPv1 and DPv0 terminology here? Why do we do DPv1 stuff in the DPv0 branch?

Also pardon my ignorance: I thought DPv0 does not exist? What is this special case?